### ISSUE TITLE
Feature Request: show or output the path to the currently configured backend

### ISSUE THREAD

**Issue by erikpaasonen** [hashicorp/terraform#19016]
Hello, this is a minor enhancement suggestion to logging of actual values that would be of most help during troubleshooting situations, especially in complex environments where the backend configuration is pieced together from several different places.

Yes it's possible to go piece the code together to see where Terraform *should be* pointing, but I'd like to suggest a stdout line that can confirm where it actually *is* pointing.
### Current Terraform Version
Terraform v0.11.8
### Use-cases
* For use as a confirmation of the resultant path to the `.tfstate` file after running `terraform init` with multiple `-backend-config=` clauses in a CI pipeline
* Troubleshooting such a CI pipeline where something went wrong with the backend config
* Troubleshooting scenario where a `terraform plan` seems to want to create every single resource and seems to have no knowledge of previously created resources
* Additional layer of safety when many repos are in the middle of being migrated from one backend to another
### Attempted Solutions
The absence of a log output line item that confirms the exact path where Terraform is looking for its state file means we have to guess where we think it should be configured to, and check that.

In our case today we were troubleshooting an S3 backend that shouldn't have changed. We validated the S3 path and accessibility of the state file in the proper path. The inclusion of multiple `-backend-config=` lines in the CI pipeline can also make it tricky to piece together the full path to the state file. In the end we found the backend block of the `terraform` resource had been commented out, thus the runtime was creating a new workspace and state file in the local directory on the build server.
### Proposal
At least two ways to output the currently-configured backend path:
1. Add a line of output to the `terraform init` command displaying the path, e.g. `./terraform.tfstate` or `s3://my-bucket-name/my/bucket/prefix/terraform.tfstate`
2. Introduce a new terraform command similar to `terraform show` that yields the full path to the state file as currently configured, dependent on a successful `init` of course

**Comment by JustinGrote** [comment]
I'd like the same to be accessible as an output from the terraform resource, e.g. terraform.tfstate.path. That way I can tag devices created with Terraform with a TERRAFORMSTATEPATH tag or somethign that shows the state, so we know which state file "manages" the terraform. We do this manually currently by combining strings and it's not the greatest.

**Comment by treksler** [comment]
i am actually quite surprised that terraform show does not show the current backend configuration. kind of like this

terraform show 
terraform show config
terraform show state <path>
terraform show outputs

**Comment by KyleKotowick** [comment]
@JustinGrote I've created a module that retrieves all backend configuration data, [Invicton-Labs/backend-config/null](https://registry.terraform.io/modules/Invicton-Labs/backend-config/null/latest). It retrieves all of the data in the plan stage.

**Comment by andy-imt** [comment]
+1 on this issue

When receiving errors like this:

```
│ Error: Backend configuration changed
│
│ A change in the backend configuration has been detected, which may require
│ migrating existing state.
│
│ If you wish to attempt automatic migration of the state, use "terraform
│ init -migrate-state".
│ If you wish to store the current configuration with no changes to the
│ state, use "terraform init -reconfigure".
```

The best that even a TF_LOG=TRACE will tell you is:
```
2022-11-16T15:48:55.210-0500 [TRACE] Meta.Backend: working directory was previously initialized for "s3" backend
2022-11-16T15:48:55.210-0500 [TRACE] backendConfigNeedsMigration: configuration values have changed, so migration is required
2022-11-16T15:48:55.210-0500 [TRACE] Meta.Backend: backend configuration has changed (from type "s3" to type "s3")
```

I would really like to be able to verify conclusively what the old and new backend data is before I tell it to change anything.

**Comment by mhemken-vts** [comment]
Yeah this would really help me figure out how to call a particular remote state without having to go into full detective mode.

**Comment by zioalex** [comment]
HI there,
any update on this topic?

**Comment by cibinmathew** [comment]
+1 for this feature request

**Comment by midokura-agustin** [comment]
Info is in the file `.terraform/terraform.state`:
```
$ jq < .terraform/terraform.tfstate .backend
{
  "type": "azurerm",
  "config": {
    "access_key": null,
    "client_certificate_password": null,
    "client_certificate_path": null,
    "client_id": null,
    "client_secret": null,
    "container_name": "###REDACTED###",
    "endpoint": null,
    "environment": null,
    "key": "###REDACTED###",
    "metadata_host": null,
    "msi_endpoint": null,
    "oidc_request_token": null,
    "oidc_request_url": null,
    "oidc_token": null,
    "oidc_token_file_path": null,
    "resource_group_name": "###REDACTED###",
    "sas_token": null,
    "snapshot": null,
    "storage_account_name": "###REDACTED###",
    "subscription_id": "###REDACTED###",
    "tenant_id": null,
    "use_azuread_auth": null,
    "use_msi": null,
    "use_oidc": null
  },
  "hash": ###REDACTED###
}
```

**Comment by vbgcwood** [comment]
Would like to add, this feature would be nice to have for certain use cases still.

If you want to see only the values that would be configured explicitly via the terraform backend block, which also may be overridden via cli args, the cli command could provide a safe way to log those values. Otherwise, interacting with tf state when the goal would be to log something is a bit uncomfortable.

**Comment by jefffroman-adaptsys** [comment]
Being able to view the configuration _in the state file_ isn't helpful, if what you're trying to do is determine which state file is being used, per the backend configuration (S3, local, etc.)
